docs/transpile/custom-transpiler-pass.ipynb

```diff
@@ -0,0 +1,352 @@
+      "text/plain": [
+       "<PIL.PngImagePlugin.PngImageFile image mode=RGBA size=305x466>"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from qiskit.converters import circuit_to_dag\n",
+    "from qiskit.tools.visualization import dag_drawer\n",
+    "dag = circuit_to_dag(circ)\n",
+    "dag_drawer(dag)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Transpiler Passes\n",
+    "\n",
+    "Transpiler passes are classified into one of two classes: Analysis passes and Transformation passes. Passes in general work with the Directed Acyclic Graph (DAG) **TODO: insert link here** and the `property_set`, a dictionary-like object for storing properties determined by analysis passes. Analysis passes work with the DAG, but are not allowed to modify it. This contrasts with transformation passes, which do modify the DAG, and can read (but not write to) `property_set`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Internals\n",
+    "\n",
+    "**TODO: put something here, but I do not know what**"
```

Comment:
I'll send you some notebooks and slide decks I have looking at this which might be useful for this.

Reply:
Ah yes I think we are covering those things in other docs.